example/server/main.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/cloudwego/kitex-examples/kitex_gen/api"
+	"github.com/cloudwego/kitex-examples/kitex_gen/api/echo"
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/server"
+	zookeeperServer "github.com/kitex-contrib/config-zookeeper/server"
+	"github.com/kitex-contrib/config-zookeeper/zookeeper"
+)
+
+var _ api.Echo = &EchoImpl{}
+
+// EchoImpl implements the last service interface defined in the IDL.
+type EchoImpl struct{}
+
+// Echo implements the Echo interface.
+func (s *EchoImpl) Echo(ctx context.Context, req *api.Request) (resp *api.Response, err error) {
+	klog.Info("echo called")
+	return &api.Response{Message: req.Message}, nil
+}
+
+func main() {
+	klog.SetLevel(klog.LevelDebug)
+	zookeeperClient, err := zookeeper.NewClient(zookeeper.Options{})
+	if err != nil {
+		panic(err)
+	}
+	serviceName := "ServiceName"
+	svr := echo.NewServer(
+		new(EchoImpl),
+		server.WithServerBasicInfo(&rpcinfo.EndpointBasicInfo{ServiceName: serviceName}),
+		server.WithSuite(zookeeperServer.NewSuite(serviceName, zookeeperClient)),
```

Comment:
Is this tested with real kitex client? Have you produced the limited cases (qps limit and/or connection limit) by changing the config in zk?

Reply:
> Is this tested with real kitex client? Have you produced the limited cases (qps limit and/or connection limit) by changing the config in zk?

I spent some time testing, and I didn't find any problems